File: examserver/src/main/java/com/exam/services/UserService.java
package com.exam.services;

import java.util.List;
import java.util.Set;

import com.exam.models.User;
import com.exam.models.UserRole;

public interface UserService {
    public User createUser(User user, Set<UserRole> userRoles) throws Exception;

    public List<User> getAllUsers() throws Exception;

    public User getUser(String userName) throws Exception;

    public void deleteUserByUserId(Long userId);

    public void deleteUserByUserName(String userName);
}

File: FrontEnd/examportal/src/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  constructor(private http: HttpClient) {}

  //
  //
  // get all users
  public getAllUsers() {
    let url = `${baseUrl}/user/getAllUsers`;
    return this.http.get(url);
  }

  //
  //
  // get user by username
  public getUserByUsername(username: String) {
    let url = `${baseUrl}/user/getUser/${username}`;
    return this.http.get(url);
  }

  //
  //
  // create user
  public createUser(user: any) {
    let url = `${baseUrl}/user/createUser`;
    return this.http.post(url, user);
  }

  //
  //
  // update user details
  public updateUserDetails(userId: any, userDetail: any) {
    let url = `${baseUrl}/user/updateUser/${userId}`;
    return this.http.put(url, userDetail);
  }
}
